.github/workflows/optimize-ci.yml

```diff
@@ -341,92 +341,13 @@ jobs:
-    if: github.event.pull_request.state != 'closed' && (contains( github.event.label.name, 'opensearch') || contains( github.event.pull_request.labels.*.name, 'opensearch')) && (needs.detect-changes.outputs.backend-changes == 'true')
-    timeout-minutes: 120
-    strategy:
-      fail-fast: false
-      matrix:
-        includedGroups: [ 'reportEvaluation', '' ]
-        include:
-          - includedGroups: ''
-            excludedGroups: 'ccsm-test,import,reportEvaluation'
-            profiles: 'it'
-          - includedGroups: 'reportEvaluation'
-            excludedGroups: ''
-            profiles: 'it'
-    steps:
-      - name: Checkout repository
-        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
-
-      - name: Fetch main branch
-        run: git fetch origin main:refs/remote/origin/main
-
-      - name: Setup Maven
-        uses: ./.github/actions/setup-maven
-        with:
-          maven-cache-key-modifier: optimize-tests
-          secrets: ${{ toJSON(secrets) }}
-
-      - name: Login to Harbor registry
-        uses: ./.github/actions/login-registry
-        with:
-          secrets: ${{ toJSON(secrets) }}
-
-      - name: "Read Java / Version Info"
-        id: "pom-info"
-        uses: YunaBraska/java-info-action@main
-        with:
-          work-dir: ./optimize
-
-      - name: Expose Parsed OpenSearch Version
-        run: echo "OPENSEARCH_VERSION=${{ steps.pom-info.outputs.x_opensearch_test_version }}" >> "$GITHUB_ENV"
-
-      - name: Start OpenSearch
-        uses: ./.github/actions/compose
-        with:
-          compose_file: .github/actions/compose/docker-compose.opensearch.yml
-          project_name: opensearch
-        env:
-          OPENSEARCH_VERSION: ${{ env.OPENSEARCH_VERSION }}
-          OPENSEARCH_JVM_MEMORY: 16
-          OPENSEARCH_HTTP_PORT: 9205
-
-      - name: Verify integration
-        uses: ./.github/actions/run-maven
-        env:
-          LIMITS_CPU: 8
-        with:
-          threads: 8
-          parameters: >
-            verify -Dit.test.excludedGroups=${{ matrix.excludedGroups }}
-            -Dit.test.includedGroups=${{ matrix.includedGroups }}
-            -Dskip.docker -Dskip.fe.build -P${{ matrix.profiles }}
-            -Dfailsafe.rerunFailingTestsCount=2
-            -Ddatabase.type=opensearch -pl optimize/backend -am
-
-      - name: Upload Test Results
-        if: always()
-        uses: ./.github/actions/collect-test-artifacts
-        with:
-          name: integration-test-opensearch-dev-${{ matrix.includedGroups }}-junit
-          path: |
-            **/failsafe-reports/**/*.xml
-
-      - name: Docker log dump
-        uses: ./.github/actions/docker-logs
-        if: always()
-        with:
-          archive_name: integration-tests-opensearch-dev-${{ matrix.includedGroups }}-docker
-
   upgrade-tests:
     name: Upgrade
-    runs-on: gcp-core-8-default
+    runs-on: gcp-core-16-default
```

Comment:
Is this because Opensearch requires more resources?

Reply:
I remember we were having instability with pipelines that run using a matrix one day and I had understood that this one core is used to run all pipelines in the matrix, which means that by running 2 pipelines instead of one I am doubling my usage. @cmur2 Is my description accurate?

Reply:
Looking at your code changes for this `upgrade-tests` job I see two things: doubling the runner capacity (8 -> 16 cores) and introducing two matrix jobs (1 -> 2 jobs each run with a runner of 16 cores). Is that what you wanted?

In practice this will increase the resource usage & cost by x4 because now we have two 2 jobs each on 16 cores and previously had 1 on 8 cores.

Reply:
Ah ok, so if it's separate jobs I will revert it to core-8 then, as there is no reason to increase the resources. Thanks @cmur2 !